index.js

```diff
@@ -183,79 +190,151 @@ const plugin = (opts = {}) => {
         });
 
         if (hasNotSelector) {
+          debugUtils.log('Skipping - already has not selector', atRule);
           atRule[processed] = true;
           return;
         }
 
         const conditions = mediaProcessor.getMediaConditions(atRule);
+        debugUtils.log(`Extracted conditions: ${JSON.stringify(conditions)}`, atRule);
+
         if (conditions.length > 0) {
           debugUtils.stats.mediaQueriesProcessed++;
 
-          // Create container version first
           const containerConditions =
             mediaProcessor.convertToContainerConditions(conditions);
 
+          debugUtils.log(`Container conditions: ${containerConditions}`, atRule);
+
           if (containerConditions) {
+            debugUtils.log('Creating container query...', atRule);
+
             const containerQuery = new helpers.AtRule({
               name: 'container',
               params: containerConditions,
               source: atRule.source,
               from: helpers.result.opts.from
             });
 
-            // Clone and process rules for container query - keep selectors clean
-            atRule.walkRules(rule => {
-              const containerRule = rule.clone({
-                source: rule.source,
-                from: helpers.result.opts.from,
-                selector: selectorHelper.updateBodySelectors(
-                  rule.selector,
-                  [ containerBodySelector ]
-                )
+            // For nested media queries
+            // For nested media queries
+            if (isNested) {
+              debugUtils.log('Processing nested media query declarations...', atRule);
+
+              atRule.each(node => {
+                if (node.type === 'decl') {
+                  debugUtils.log(`  Processing declaration: ${node.prop}: ${node.value}`, atRule);
+
+                  const containerDecl = node.clone({
+                    source: node.source,
+                    from: helpers.result.opts.from
+                  });
+
+                  // Convert viewport units if needed
+                  let value = containerDecl.value;
```

Comment:
Why defining value outside of the scope of the `if` statement if you use it only inside it. Should be defined inside the `if`.